### ISSUE TITLE
[Question]: torch-geometric version

### ISSUE THREAD

**Issue by whangdo** [will-leeson/sibyl#1]
Greetings, recently, I tried to train gnn in sibyl. However, an error below occurred.
```
./run_study.sh -d ICSE -j 1
/usr/bin/nvidia-smi
You have NVIDIA Support
We can proceed with training.

==============================================
              Building Graphs
==============================================


==============================================
             Training Network
==============================================

Traceback (most recent call last):
  File "/home/hdy/smtfuzz/sibyl/src/networks/smtTrainer.py", line 67, in <module>
    train_set = SMTDataset(trainLabels, dataLoc, args.edge_sets, tracks, args.cache)
TypeError: Can't instantiate abstract class SMTDataset with abstract methods get, len
Training appears to have failed
Please check error messages
```
As far as I know, it is related to `torch-geometric`. So, I tried to fit version of the library however, there was no `1.10 version`.
```
pip3 install torch-geometric==1.10
Defaulting to user installation because normal site-packages is not writeable
ERROR: Could not find a version that satisfies the requirement torch-geometric==1.10 (from versions: 0.1.1, 0.2.0, 0.3.0, 0.3.1, 1.0.0, 1.0.1, 1.0.2, 1.0.3, 1.1.0, 1.1.1, 1.1.2, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.3.2, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.6.0, 1.6.1, 1.6.2, 1.6.3, 1.7.0, 1.7.1, 1.7.2, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.1.0, 2.1.0.post1, 2.2.0, 2.3.0, 2.3.1, 2.4.0)
ERROR: No matching distribution found for torch-geometric==1.10
```
Could I know the correct version of torch-geometric?
Thanks for reading.

**Comment by will-leeson** [comment]
Hi, can you tell me what version of PyTorch and PyTorch-Geometric you are using?

**Comment by will-leeson** [comment]
It looks like installing prior version of torch-geometric is a bit more complicated than normal pip packages. Here is how they suggest:
```
pip install pyg_lib torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-${TORCH}+${CUDA}.html
```
Where ${TORCH} is your torch is your torch version, and ${CUDA} is your CUDA version.
Here are valid 1.10 versions

![image](https://github.com/will-leeson/sibyl/assets/21344599/e7cbae8f-2839-42db-abf9-7ab19ca94e92)

**Comment by whangdo** [comment]
@will-leeson Thanks for kind answer! Versions of torch and torch-geometric are as below.
```
root@27daf000a6f8:/workspace/sibyl# pip3 show torch
Name: torch
Version: 1.10.0
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /opt/conda/lib/python3.7/site-packages
Requires: typing-extensions
Required-by: torchvision, torchtext, torchelastic

root@27daf000a6f8:/workspace/sibyl# pip3 show torch-geometric
Name: torch-geometric
Version: 2.3.1
Summary: Graph Neural Network Library for PyTorch
Home-page: None
Author: None
Author-email: Matthias Fey <matthias@pyg.org>
License: None
Location: /opt/conda/lib/python3.7/site-packages
Requires: pyparsing, tqdm, scikit-learn, requests, jinja2, psutil, scipy, numpy
Required-by:
```
To be specific, version of torch is `'1.10.0+cu102'`
After receiving your reply, I tried to generate environment as similar to the configs as possible using nvidia-docker. However, I cannot execute `sibyl` still. 
Also, `pip install pyg_lib torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-1.10.0+cu102.html` makes errors. `torch-cluster, torch-scatter, torch-sparse, torch-spline-conv, pyglib` would not be installed.

**Comment by whangdo** [comment]
@will-leeson 
if you are okay, could you upload your dockerfile on dockerhub? If so, I guess I can generate environment to execute sibyl with nvidia-docker :)

**Comment by will-leeson** [comment]
It looks like PyTorch geometric requires PyTorch 11.1 or greater. I'm going to see how involved it will be to update things. If it's quick, then I will go through with that. If it will take a while, I'll go through the dockerhub process while I do the updates

**Comment by will-leeson** [comment]
@fwangdo I believe it should be fixed now. When PyG updated to >2, they changed how datasets work. It should be fixed now. I was able to start training on pyg 2.4.0, so it should work for 2.3.1. I did use a newer version of Torch, but that shouldn't affect it. Let me know if you have any other issues

**Comment by whangdo** [comment]
@will-leeson Thanks for your kindness. I checked that GNN worked! 
However, I had to do something that you did not mention to succeed in training GNN. I am not sure whether It was from my own environment or not but I wrote what I had to do cuz it might be helpful.

1. I had to write script code to generate `npz` file. `build_graphs.sh` does not work in my env.
2. I had to use `torch==1.13.1`(it might be caused by my cuda version)

Thank you again :)
